From 64387d5598e5c72a3d8fef0b31444c02df2f2cb2 Mon Sep 17 00:00:00 2001
From: Cyril Brulebois <kibi@debian.org>
Date: Tue, 23 Nov 2010 19:06:10 +0100
Subject: [PATCH] Revert "dropped goodmap and badmap for apple macbooks, needed only for old broken kernels, b.fd.o#9095"

Since kernels may not have a proper fix, keep those options available.

This reverts commit 222afd3220e9c920bef41bc2e95886c5d4414d5b.

Conflicts:

	ChangeLog

Signed-off-by: Cyril Brulebois <kibi@debian.org>
---
 keycodes/macintosh  |   10 ++++++++++
 rules/base.m_k.part |    1 +
 rules/base.o_k.part |    2 ++
 3 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/keycodes/macintosh b/keycodes/macintosh
index d41ac4b..d1e0be8 100644
--- a/keycodes/macintosh
+++ b/keycodes/macintosh
@@ -36,6 +36,16 @@ default xkb_keycodes "macintosh" {
    
 };
 
+xkb_keycodes "badmap" {
+    <TLDE> =  94;
+    <LSGT> =  49;
+};
+
+xkb_keycodes "goodmap" {
+    <TLDE> =  49;
+    <LSGT> =  94;
+};
+
 xkb_keycodes "old" {
 
     minimum= 8;
diff --git a/rules/base.m_k.part b/rules/base.m_k.part
index dd0c391..794994c 100644
--- a/rules/base.m_k.part
+++ b/rules/base.m_k.part
@@ -9,6 +9,7 @@
  $applealu	=	macintosh(alukbd)
   macintosh_hhk	=	macintosh(hhk)
   macintosh_old	=	macintosh(old)
+ $macbooks	=	macintosh+macintosh(goodmap)
  $macs   	=	macintosh
   thinkpadz60	=	xfree86(thinkpadz60)
   *		=	xfree86
diff --git a/rules/base.o_k.part b/rules/base.o_k.part
index e69de29..53d2879 100644
--- a/rules/base.o_k.part
+++ b/rules/base.o_k.part
@@ -0,0 +1,2 @@
+  apple:badmap		=	+macintosh(badmap)
+  apple:goodmap		=	+macintosh(goodmap)
-- 
1.7.2.3

